### ISSUE TITLE
AST changes when formatting type applications

### ISSUE THREAD

**Issue by ocharles** [fourmolu/fourmolu#211]
The following

```haskell
foo = bar @'Bar
```

Gets reformatted to 

```haskell
foo = bar @ 'Bar
```

I can't reproduce this on https://ormolu-live.tweag.io/, so think it might be a Fourmolu specific bug.

**Comment by georgefst** [comment]
Thanks. That's odd. I can't think of anything relevant that we alter from Ormolu here. And I'm quite surprised this isn't caught be our test suite.

**Comment by georgefst** [comment]
I've tried Ormolu 0.5 and Fourmolu 0.7, both built with the same `ghc-lib-parser` and can observe a difference.

**Comment by ocharles** [comment]
@georgefst Just to clarify, do you mean you can reproduce this bug with Ormolu as well? (Also, hi, isn't it funny that I was adamant I hate autoformatting at ZuriHac, and here I am weeks later, auto-formatting :stuck_out_tongue_closed_eyes:)

**Comment by georgefst** [comment]
I mean I can reproduce with Fourmolu but not Ormolu.

@brandonchinn178 Any ideas? A bad merge somewhere?

**Comment by brandonchinn178** [comment]
Possibly. I looked at Meat/Value.hs at the AppType section, but it looked the same on both ends. Will probably need good ol' debugging work

**Comment by georgefst** [comment]
> Also, hi, isn't it funny that I was adamant I hate autoformatting at ZuriHac, and here I am weeks later, auto-formatting

Ha, yeah, glad to have changed your mind.

I don't think I actually caught your name at the time, and only realised who you were a few days later when something about `reactive-banana` came up and I realised you were the maintainer (I recognise you from a few things e.g. we use your Nix Buildkite plugin at work)!

**Comment by ocharles** [comment]
Yes, ZuriHac needs bigger name badges :laughing:

**Comment by ocharles** [comment]
If it helps, this only happens when using quoted constructors - e.g., `foo @True` is fine, `foo @'True` is problematic.

**Comment by SeungheonOh** [comment]
Also I found out putting a ticked type inside of parenthesis makes parser happy again.

```hs
-- Unhappy parser
id @'Int

-- Happy parser
id @('Int)
```

It's quite annoying with `-Wunticked-promoted-constructors`... I hope it get fixed.

**Comment by elland** [comment]
Is this still being worked on?

**Comment by brandonchinn178** [comment]
No one's working on it yet. Feel free to open a PR

**Comment by brandonchinn178** [comment]
Looks like a bad merge 😢 

Ormolu fixed this back in 0.1.4.0: https://github.com/tweag/ormolu/commit/286afb7ed6bd2985339c617cbdeae5c8ab4b8353#diff-88777f1040ac8f0f2e2903f2db73e3604774239dff0453bc9fcb68c6ceab1f66

I merged this into Fourmolu as one of my first contributions, and this was before Fourmolu was systematically merging Ormolu changes in, so it was a huge change that I probably overlooked this conflict.
https://github.com/fourmolu/fourmolu/pull/106/commits/948e32deff1f18afe050e6eea3ab37cac1f9d948

I'll make the change (and add a regression test cuz why not)

**Comment by SeungheonOh** [comment]
Thank you for the fix! Can I know when this bugfix will be released?

**Comment by brandonchinn178** [comment]
I'm hoping to get https://github.com/fourmolu/fourmolu/pull/245 merged in for the next release, so maybe in the next week or so?

**Comment by brandonchinn178** [comment]
Released v0.9.0.0

**Comment by SeungheonOh** [comment]
Awesome! a
